src/core/server/elasticsearch/elasticsearch_service.test.ts

```diff
@@ -412,15 +414,18 @@ describe('#stop', () => {
 
     const setupContract = await elasticsearchService.setup(setupDeps);
 
-    setupContract.esNodesCompatibility$.subscribe(async () => {
-      expect(mockedClient.nodes.info).toHaveBeenCalledTimes(1);
-      await delay(10);
-      expect(mockedClient.nodes.info).toHaveBeenCalledTimes(2);
-
-      await elasticsearchService.stop();
-      await delay(100);
-      expect(mockedClient.nodes.info).toHaveBeenCalledTimes(2);
-      done();
+    return new Promise<void>((done) => {
+      const subscription = setupContract.esNodesCompatibility$.subscribe(async () => {
+        expect(mockedClient.nodes.info).toHaveBeenCalledTimes(1);
+        await delay(10);
+        expect(mockedClient.nodes.info).toHaveBeenCalledTimes(2);
+
+        await elasticsearchService.stop();
+        await delay(100);
+        expect(mockedClient.nodes.info).toHaveBeenCalledTimes(2);
+        subscription.unsubscribe();
+        done();
+      });
```

Comment:
The `.subscribe(next)` callback doesn't support promises, so the promise created by the async function passed to `subscribe()` will be ignored, triggering a unhandled promise rejection and process termination if these assertions ever fails To make sure that errors produced by `expect()` are propagated properly you should keep assertions in the observable using the proper operator. In this case, `concatMap` should be used because it will serially call the passed function for each value produced by `setupContract.esNodesCompatibility$`, wait for the promise to resolve, then produce the resolved value on the observable (undefined in this case because the async function has no return value), which will then trigger the `firstValueFrom()` helper and cause the observable to be unsubscribed.

```suggestion
    await firstValueFrom(
      setupContract.esNodesCompatibility$.pipe(
        Rx.concatMap(async () => {
          expect(mockedClient.nodes.info).toHaveBeenCalledTimes(1);
          await delay(10);
          expect(mockedClient.nodes.info).toHaveBeenCalledTimes(2);

          await elasticsearchService.stop();
          await delay(100);
          expect(mockedClient.nodes.info).toHaveBeenCalledTimes(2);
        })
      )
    )

```

Reply:
TIL 😲